### PR TITLE
2186/2187/2190: 319 Code Scanning Alert Fixes and Two LGTM Fixes

### LINKED ISSUE #2186: [Quality] Use of default toString()
Default toString used in a manner that isn't useful:

https://github.com/MegaMek/mekhq/security/code-scanning?query=tool%3ACodeQL+is%3Aopen+id%3Ajava%2Fcall-to-object-tostring

### LINKED ISSUE #2187: [Quality] Possible confusion of local and field
Poorly named locals shadowing fields that should be updated:

https://github.com/MegaMek/mekhq/security/code-scanning?query=id%3Ajava%2Flocal-shadows-field

### LINKED ISSUE #2190: [Quality] Useless toString on a string
Instances of Strings having toString called on them: 

https://github.com/MegaMek/mekhq/security/code-scanning?query=id%3Ajava%2Fuseless-tostring-call

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #2186, fixes #2187, and fixes #2190.

The one newly detected alert isn't easily fixable, as it's our entity setup.

To Review:
I separated alerts into their own commits, while FactionHints and JSuggestField changes were separated out as they were more significant than the base alert fix (the former because the alert was actually because of a few issues, the latter because of a bunch of other small formatting setups).
Whitespace differences off, as I fixed tab spacing in JSuggestField and a bunch of other whitespace issues as I noticed them.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/MercRosterAccess.java [hunk 1]
```diff
@@ -351,6 +358,12 @@ private void writeBasicData() {
                         preparedStatement.setInt(2, role.ordinal());
                         preparedStatement.executeUpdate();
                         break;
+                    case ASTECH:
+                    case MEDIC:
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/MercRosterAccess.java:362]: Wait, forgot these two actual have skills attached.

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/MercRosterAccess.java]: And fixed.

File: MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java [hunk 2]
```diff
@@ -1159,8 +1157,8 @@ private void addGroup(IOptionGroup group, GridBagLayout gridBag, GridBagConstrai
         panOptions.add(groupLabel);
     }
 
-    private void addOption(IOption option, GridBagLayout gridBag, GridBagConstraints c, boolean editable) {
-        DialogOptionComponent optionComp = new DialogOptionComponent(this, option, editable);
+    private void addOption(IOption option, GridBagLayout gridBag, GridBagConstraints c) {
+        DialogOptionComponent optionComp = new DialogOptionComponent(this, option, true);
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java:1161]: I assume the editable parameter was always passed in as true?

**Review by NickAragua — COMMENTED** [review]
Some thoughts on it. Good to reduce warning spam overall.

File: MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java (hunk 2 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java:1161]: Yup

File: MekHQ/src/mekhq/campaign/personnel/SkillPrereq.java [hunk 3]
```diff
@@ -207,16 +204,13 @@ public static SkillPrereq generateInstanceFromXML(Node wn) {
...
-                    if(null != SkillType.getType(skillName)) {
+                    if (null != SkillType.getType(skillName)) {
                         retVal.addPrereq(skillName, level);
                     }
                 }
             }
         } catch (Exception ex) {
-            // Errrr, apparently either the class name was invalid...
-            // Or the listed name doesn't exist.
-            // Doh!
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/SkillPrereq.java:218]: I normally do... but this one is duplicated so many times it has become annoying and slows down my parsing

File: MekHQ/src/mekhq/campaign/personnel/SkillType.java [hunk 4]
```diff
@@ -465,10 +460,7 @@ public static void generateSeparateInstanceFromXML(Node wn, Hashtable<String, Sk
 
             hash.put(retVal.name, retVal);
         } catch (Exception ex) {
-            // Errrr, apparently either the class name was invalid...
-            // Or the listed name doesn't exist.
-            // Doh!
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/SkillType.java:470]: Part.java is one that can be easily left

File: MekHQ/src/mekhq/campaign/personnel/Bloodname.java [hunk 5]
```diff
@@ -492,17 +491,29 @@ public Fraction(Fraction f) {
         }
 
         @Override
-        public Object clone() {
-            return new Fraction(this);
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/Bloodname.java:496]: I never even paid attention to the class, was just trying to fix the issues... and this is not the only place we have private Fraction classes.

File: MekHQ/src/mekhq/campaign/MercRosterAccess.java [hunk 6]
```diff
@@ -270,6 +270,13 @@ private void writeBasicData() {
                         preparedStatement.executeUpdate();
                         equipment = 1;
                         break;
+                    case PROTOMECH_PILOT:
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/MercRosterAccess.java:273]: No clue, but I doubt it.

File: MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java [hunk 7]
```diff
@@ -301,14 +301,10 @@ public void setTimeLimitType(TimeLimitType timeLimitType) {
...
     public String getTimeLimitString() {
-        String timeLimitString = 
-                timeLimitType == TimeLimitType.None ? 
-                    "" :
-                    String.format("%s %d turns", isTimeLimitAtMost() ? " within at most" : " for at least", getTimeLimit());
-        
-        return timeLimitString;
-    }
-    
+        return (timeLimitType == TimeLimitType.None) ? "" :
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: Ternaries within ternaries aren't the easiest but are far from the worst. I could simplify it down by swapping to a switch?

[the remark this answers is not in the document]
Comment (NickAragua) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: Yeah, that'll work.

[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: It's the enum itself that is the problem, namely that it should have the `timeLimitAtMost` variable merged into it. Would you be good with merging this in, then I'll handle the enum update separately?

[the remark this answers is not in the document]
Comment (NickAragua) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: Yeah, sounds good.

[the remark this answers is not in the document]
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: Anything else to handle then?

[the remark this answers is not in the document]
Comment (NickAragua) [reply MekHQ/src/mekhq/campaign/mission/ScenarioObjective.java:304]: Nah

**Review by NickAragua — APPROVED** [review]

**Pull request merged**